### PR TITLE
Adding in progress status

### PR DESCRIPTION
**Pull request by alaGarbey675**
This pull request introduces an "In Progress" status feature. This feature allows users to view tasks that are currently in progress, enhancing visibility and tracking of ongoing jobs.

### CODE REVIEW

File: pkg/webhook_test.go [hunk 1]
```diff
@@ -32,7 +32,7 @@ func TestJobRunWebhookCall(t *testing.T) {
 
 	// test generic webhook
 	jr := JobRun{
-		Status:      0,
+		Status:      Int(0),
```
Comment (bart6114) [inline pkg/webhook_test.go:35]: same

**Review by bart6114 — COMMENTED** [review]
Nice job, great refacs & adding of comments as well!
Added a few remarks here and there, all in all very minor ones.

File: pkg/job.go [hunk 2]
```diff
@@ -161,37 +198,53 @@ func (j *JobSpec) execCommand(trigger string) JobRun {
...
+		exitCode := -1
+		j.log.Warn().Str("job", j.Name).Str("trigger", trigger).Int("exitcode", exitCode).Err(err).Msg("job unable to start")
+
+		// Also send this to terminal output
 		_, err = w.Write([]byte(fmt.Sprintf("job unable to start: %v", err.Error())))
 		if err != nil {
 			j.log.Debug().Str("job", j.Name).Err(err).Msg("can't write to log buffer")
 		}
 
+		jr.Status = Int(exitCode) // Set the exit code in the job result
```
[the remark this answers is not in the document]
Comment (alaGarbey675) [reply pkg/job.go:222]: we don't have a test case specifically covering errors in cmd.Start. I will add a test case to ensure that we capture this scenario.

File: pkg/job.go [hunk 3]
```diff
@@ -161,37 +198,53 @@ func (j *JobSpec) execCommand(trigger string) JobRun {
...
+			exitCode := -1
+			j.log.Error().Str("job", j.Name).Err(err).Msg("unexpected error during command execution")
+			jr.Status = Int(exitCode)
+			return jr
 		}
-
-		return jr
+	} else {
+		// No error, command exited successfully
+		jr.Status = Int(0) // Command succeeded, set exit code 0
```
[the remark this answers is not in the document]
Comment (alaGarbey675) [reply pkg/job.go:242]: // Global status constants
const (
	StatusOK    int = 0
	StatusError int = -1
)

File: pkg/job.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (alaGarbey675) [reply pkg/job.go:222]: ### **added two func :**  

**TestExecCommandStartError :**

This function simulates a scenario where the command's execution fails to start. It creates a JobRun instance and a JobSpec with a command that does not exist ("nonexistent-command"). 

**TestExecCommandExitError :**

This function simulates a scenario where the command starts but exits with an error (a non-zero exit code). It also creates a JobRun instance and a JobSpec, but this time the command is "false", which always exits with code 1

**Comment by bart6114** [comment]
Nice one @alaGarbey675 ! 🥳 

Could you also merge and test the open Dependabot PRs into this branch? And then we do one big merge into main. Doing it one by one will trigger a new release every time.

**Comment by alaGarbey675** [comment]
All Dependabot PRs have now been combined into this one.

**Pull request merged**